### PR TITLE
API - Medicines (Entity, Repository, DTOs, Mapper, Service, Controlle…

### PR DESCRIPTION
**Pull request by york-rcornidez**
…r, Tests) and Updated API Reference in README

Updated:
- README - Medicine API reference linked to the ApiReference dir.

Created:
- Medicine Entity
- Medicine Repository
- Medicine Request/Response DTOs
- Medicine Mapper and Tests
- Medicine Service and Tests
- Medicine Controller and Tests

### CODE REVIEW

File: src/main/java/york/pharmacy/medicines/MedicineController.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    @PutMapping("/{id}")
+    public ResponseEntity<MedicineResponse> updateMedicine(@PathVariable Long id, @Valid @RequestBody MedicineRequest medicineRequest) {
+        MedicineResponse medicineResponse = medicineService.updateMedicine(id, medicineRequest);
+        return new ResponseEntity<>(medicineResponse, HttpStatus.OK);
+    }
+
+    // Delete a medicine by its ID
+    @DeleteMapping("/{id}")
+    public ResponseEntity<Void> deleteMedicine(@PathVariable Long id) {
+        medicineService.deleteMedicine(id);
```
Comment (york-egraham) [inline src/main/java/york/pharmacy/medicines/MedicineController.java:58]: We should handle the exception thrown by the `deleteMedicine` service method.

Comment (york-rcornidez) [reply src/main/java/york/pharmacy/medicines/MedicineController.java:58]: This exception handling is handled in the Global Exception Handler Class, whenever an error is thrown it catches it and responds with the appropriate status code. Specifically the not found (404) and server error (500) status codes.

File: src/main/java/york/pharmacy/medicines/Medicine.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @NotNull
+    private String name;
+
+    @NotNull
+    private String code;
```
Comment (york-egraham) [inline src/main/java/york/pharmacy/medicines/Medicine.java:29]: Is there a way to mark code as unique so the database won't store duplicates?

Comment (york-rcornidez) [reply src/main/java/york/pharmacy/medicines/Medicine.java:29]: Great point, that is important. It looks like I can implement this annotation to solve this.
```Java
@UniqueField(
        fieldName = "code",
        entityClass = Medicine.class,
        message = "Code must be unique"
 )
 ```

**Review by york-ymost — APPROVED** [review]
legit.

**Pull request merged**